### ISSUE TITLE
Step over, doesn't stop at next instruction...

### ISSUE THREAD

**Issue by jcward** [jcward/vscode-hxcpp-debug#49]
Not sure if it's related to the [breakpoint issue](https://github.com/HaxeFoundation/hxcpp-debugger/issues/17), or if it's on my end, but with the latest hxcpp (3.4.64), the step over function doesn't stop at the next line. It acts like pressing "Run" / "Continue".

**Comment by jcward** [comment]
Note that pressing "Step Over" is like pressing "Continue" -- another breakpoint will stop it.

Also note, I'm not yet handling this "ThreadLocation" message from the debugger:

![image](https://cloud.githubusercontent.com/assets/2192439/24059937/97e46c4e-0b16-11e7-8b93-52584560ca98.png)

When pressing Step Over, the output is:
```
About to press step over...
1489779367.39966: Reading 70 bytes...
1489779367.39969: {"command":"next","arguments":{"threadId":0},"type":"request","seq":8}
1489779367.3997: Got command: next
1489779367.39972: ----> Sending 67 bytes:
1489779367.39973: {"command":"next","success":true,"request_seq":8,"type":"response"}
1489779367.39976: VSCH: got cmd: Next(1)
1489779367.44982: Got message: OK
1489779367.44984: ====== UNHANDLED MESSAGE: OK
1489779367.4999: Got message: ThreadStarted(0)
1489779367.54998: Got message: ThreadCreated(3)
1489779367.55: Thread 3 created.
```

When pressing Continue, the output is:
```
About to press continue...
1489779418.32282: Reading 74 bytes...
1489779418.3229: {"command":"continue","arguments":{"threadId":0},"type":"request","seq":8}
1489779418.32293: Got command: continue
1489779418.32297: VSCH: got cmd: Continue(1)
1489779418.37301: Got message: OK
1489779418.37303: ====== UNHANDLED MESSAGE: OK
1489779418.4231: Got message: ThreadStarted(0)
1489779418.42312: ----> Sending 71 bytes:
1489779418.42313: {"command":"continue","success":true,"request_seq":8,"type":"response"}
1489779418.4732: Got message: ThreadCreated(3)
1489779418.47322: Thread 3 created.
```

Hmm, it sure seems that I'm sending the correct messages to the debugger, lemmi try a local debugger testcase.

**Comment by jcward** [comment]
Left a comment on the breakpoint issue -- I believe the `Next` command isn't being handled correctly.

**Comment by JohnSword** [comment]
hello jcward. I noticed another problem. When removing a breakpoint with the debugging session still running the breakpoint continues to break and it's not really removed (only visually in the editor). One needs to restart the app so the breakpoint is removed.

**Comment by jcward** [comment]
Hi @JohnSword -- yes, I see this behavior also, and it looks like a bug on my end. Thanks!

Notes to self:

Looking at adapter.log, it appears I'm always sending AddFileLineBreakpoint commands. I need to keep a local representation of "added" breakpoints, then remove them when the list is updated.

e.g. on adding breakpoint at line 19, vscode sends me:

```
{"command":"setBreakpoints","arguments":{"source":{"name":"/Main.hx","path":"/home/jward/dev/vscode-hxcpp-debug/test cli/Main.hx"},"lines":[19],"breakpoints":[{"line":19}],"sourceModified":false},"type":"request","seq":24}
```

And I send: `AddFileLineBreakpoint(Main.hx,19)`

Adding on 24:

```
{"body":{"breakpoints":[{"verified":true,"line":19},{"verified":true,"line":24}]},"command":"setBreakpoints","success":true,"request_seq":23,"type":"response"}
```

And I send: `AddFileLineBreakpoint(Main.hx,19)` `AddFileLineBreakpoint(Main.hx,24)`

Then removing 24:
```
{"command":"setBreakpoints","arguments":{"source":{"name":"/Main.hx","path":"/home/jward/dev/vscode-hxcpp-debug/test cli/Main.hx"},"lines":[19],"breakpoints":[{"line":19}],"sourceModified":false},"type":"request","seq":24}
```

And I send: `AddFileLineBreakpoint(Main.hx,19)`

I need to maintain state and send [DeleteFileLineBreakpoint](https://github.com/HaxeFoundation/hxcpp-debugger/blob/master/debugger/DebuggerThread.hx#L211-L212).

**Comment by jcward** [comment]
The step over issue is fixed in hxcpp 3.4.69, commit [4a2fbd](https://github.com/HaxeFoundation/hxcpp/commit/4a2fbdb4952b9e6ae5b10cb27962e8f10f5fd117)
